lib/net.js

```diff
@@ -736,9 +736,11 @@ Socket.prototype._getpeername = function() {
   if (!this._handle || !this._handle.getpeername || this.connecting) {
     return this._peername || {};
   } else if (!this._peername) {
-    this._peername = {};
+    const out = {};
+    const err = this._handle.getpeername(out);
     // FIXME(bnoordhuis) Throw when the return value is not 0?
```

Comment:
I think this TODO can be removed then? cc @bnoordhuis

Reply:
I think it can be removed. 

The getters, such as `remoteAddress` and `remotePort`, using this function currently return `undefined` when `getpeername` on the handle isn't available. We can say that `getpeername` isn't available also when the `err` is not 0. Returning `{}` looks fine in that case.

Reply:
could you update?

Reply:
Applied. Thanks!